Modules/System/SharePoint/src/SharePointClient.Codeunit.al

```diff
@@ -315,7 +340,48 @@ codeunit 9100 "SharePoint Client"
     /// <returns>True if the operation was successful; otherwise - false.</returns>
     procedure AddFileToFolder(ServerRelativeUrl: Text; FileName: Text; var FileInStream: InStream; var SharePointFile: Record "SharePoint File" temporary): Boolean
     begin
-        exit(SharePointClientImpl.AddFileToFolder(ServerRelativeUrl, FileName, FileInStream, SharePointFile));
+        exit(SharePointClientImpl.AddFileToFolder(ServerRelativeUrl, FileName, FileInStream, SharePointFile, false));
+    end;
+
+    /// <summary>
+    /// Adds a file to specific folder.
+    /// </summary>
+    /// <remarks>Does not require UI interaction.</remarks>
+    /// <param name="ServerRelativeUrl">URL of the parent folder.</param>
+    /// <param name="FileName">File name to use on SharePoint.</param>
+    /// <param name="FileInStream">File stream to upload.</param>
+    /// <param name="SharePointFile">Collection of the result (temporary record). Always one element.</param>
+    /// <param name="ListAllFields">Include metadata in results.</param>
+    /// <returns>True if the operation was successful; otherwise - false.</returns>
+    procedure AddFileToFolder(ServerRelativeUrl: Text; FileName: Text; var FileInStream: InStream; var SharePointFile: Record "SharePoint File" temporary; ListAllFields: Boolean): Boolean
```

Comment:
If I understand your code correctly then this function can raise the event `ProcessSharePointFileMetadata`?

Could you please update the documentation to include a raises information for the events, that migth be raised by this procedure?